Improve security document (#22713)
opf/openproject: docs/security-and-privacy/statement-on-security/README.md

6. A patch is returned to the reporter and awaited for confirmation, unless the fix is trivial.
7. A patch release is created, published, and distributed for all supported installations.
8. The security vulnerability is publicly disclosed on GitHub and communicated through the mailing list.

### How long does it take to fix a security issue?

All vulnerabilities are treated with the highest priority. Critical and high impact vulnerabilities are aimed to be available as an upgradable fix within 14 days after confirmation of the vulnerability. This timeline represents our targets. Actual resolution times may vary depending on the complexity of the issue and the availability of our team. 

### When is a security issue considered fixed?

A security issue is considered fixed only once the fix has been released for all supported versions affected by the issue.

### Public disclosure

Once a security issue has been properly fixed and the fix has been released, a CVE is published to publicly disclose the issue and encourage upgrades. We follow a responsible disclosure timeline:

> - After the fix is released, an embargo period begins to give administrators time to upgrade their installations before the vulnerability is made public.

Reviewer: @oliverguenther How long is the embargo?

> Any security related information will also be published on our [blog](https://www.openproject.org/blog/) and in the [release notes](../../release-notes/).

Reviewer 2: This is not really true. We did a lot of security patches and did not do any blog post. We should probably mention Github Advisoris here: 

```suggestion
Any security related information will also be published in the [release notes](../../release-notes/) and as [GitHub Security Advisories](https://github.com/opf/openproject/security/advisories).
```

### Security advisory list

> OpenProject uses GitHub to manage and publish [security advisory listings](https://github.com/opf/openproject/security/advisories). All publicly known security issues on OpenProject are reported as CVEs and can be found through these advisories. All published CVEs will also be listed on [cve.org](https://www.cve.org/CVERecord/SearchResults?query=OpenProject).

Reviewer 2: Everything we publish as advisories will also get a CVE, I guess. But for the lower CVSS scores, we often decide not to put out an advisory/CVE and instead treat this as a Bug. I wonder if we should apply that phrasing. 

Something along

>  All publicly known security issues on OpenProject with at least a medium CVSS score are reported as CVEs and can be found through these advisories.

### Email

> For direct communication regarding security matters, please contact [security@openproject.com](mailto:security@openproject.com). This channel is monitored by our security team and is the primary entry point for vulnerability reports.

Reviewer 2: Is this the primary entry point? It is one, but before it sounded like we prefer GHA. Fine with both. Maybe also rephrase as something like

> Next to GitHub advisories, you can also contact us via email ...

Providing clear reproduction steps is essential. Reports that lack sufficient detail for us to validate the issue may take significantly longer to process.

### Responsible disclosure and testing policy

Please adhere to the following rules when reporting or researching security issues:

- **Only test your own instances**: Targeting other users' instances is forbidden. Only test against your own OpenProject installation. If you need additional testing instances or other data from us, feel free to reach out.
- **No public disclosure without agreement**: No vulnerability disclosure, full, partial, or otherwise, is allowed without prior agreement from us. Please avoid submitting security issues on our public repositories before reporting them through the proper channels.
- **Redact sensitive data**: Do not include Personally Identifiable Information (PII) in your report. Redact or obfuscate PII, secrets, keys, and credentials that appear in screenshots, server responses, or other evidence.
- **Human interaction required:** Please do not report unchecked and obvious AI-generated reports, or we may have to close them without further comments.

## Criteria for security vulnerabilities

> We appreciate your time in every security report you communicate to us. There are a number of cases that can be viewed as security vulnerabilities, but for which we might either reject or not follow up with a full CVE publication. We still welcome you to reach out and discuss potential mitigations or attack vectors with us. Examples for these cases could be

Reviewer 2: We should also add a hint that only the latest released version of OpenProject should be targeted.

Reviewer 2: Should we also mention that for critical (CVSS >= 9.0) we attempt to fix the the current and the previous major version. But normally we only offer security patches for the latest released minor version

As a general guideline:

- **Critical**: Vulnerabilities with a high CVSS score that can be exploited remotely with no or low privileges, no user interaction, and high impact on confidentiality, integrity, or availability.
- **High**: Vulnerabilities with significant impact but that require some privileges, user interaction, or have high attack complexity.
- **Medium/Low**: Vulnerabilities with limited impact or that require high privileges, unusual configurations, or active user interaction to exploit.

## Security vulnerability processing

When we receive vulnerability reports from researchers or through internal identification, the following process takes place:

1. A security vulnerability is reported internally, through GitHub advisories, an active bug bounty program, or through [security@openproject.com](mailto:security@openproject.com).
2. A security engineer receives and validates the report. An internal tracking ticket is created with a checklist template on how to process the report.
3. The reporter receives a timely response with an acknowledgement of the report, further questions if present, and an estimated timeline and complexity of a potential fix.

> 4. The security engineer coordinates with the security and development team to prepare and test a fix in a private branch (using GitHub Security Advisory temporary private forks).

Reviewer 2: I would not mention the private fork here, as we do not really use this but we use our long running fork.

We should maybe mention that we will work on the fix in private and only merge it into he public repository immediately before the release and publication of the CVE

5. A GitHub advisory draft is created and a CVE is requested, if appropriate. Security researchers are invited to collaborate on the draft, if available.
6. A patch is returned to the reporter and awaited for confirmation, unless the fix is trivial.

> 7. For critical and high-severity issues, a **pre-release notification** is sent to the security mailing list at least 7 days before the planned release, including the release date and severity (but no vulnerability details).

Reviewer 2: I like this a lot!
